File: backend/app/schemas/venue.py
from datetime import time
from decimal import Decimal

from pydantic import BaseModel, Field


class VenueBase(BaseModel):
    name: str = Field(min_length=1, max_length=150)
    address: str | None = Field(default=None, max_length=300)
    phone: str | None = Field(default=None, max_length=30)
    capacity: int = Field(ge=1)

    opening_time: time | None = None
    closing_time: time | None = None

    facebook_url: str | None = Field(default=None, max_length=300)
    whatsapp_number: str | None = Field(default=None, max_length=30)
    instagram_url: str | None = Field(default=None, max_length=300)
    website_url: str | None = Field(default=None, max_length=300)

    base_price: Decimal = Field(ge=0, max_digits=10, decimal_places=2)
    bouncy_castle_cost: Decimal = Field(
        ge=0,
        max_digits=10,
        decimal_places=2,
    )
    extra_hour_price: Decimal = Field(
        ge=0,
        max_digits=10,
        decimal_places=2,
    )

    bank_name: str | None = Field(default=None, max_length=100)
    bank_account_holder: str | None = Field(default=None, max_length=150)
    bank_account_number: str | None = Field(default=None, max_length=50)
    bank_clabe: str | None = Field(default=None, max_length=30)

    facebook_response_message: str | None = None
    banking_information_message: str | None = None
    location_message: str | None = None
    venue_rules_message: str | None = None
    payment_reminder_message: str | None = None
    thank_you_message: str | None = None
    general_notes: str | None = None


class VenueUpdate(VenueBase):
    pass


class VenueRead(VenueBase):
    id: int

    model_config = {"from_attributes": True}

File: frontend/src/types/venue.ts
export interface Venue {
  id: number;

  name: string;
  address: string | null;
  phone: string | null;
  capacity: number;

  opening_time: string | null;
  closing_time: string | null;

  facebook_url: string | null;
  whatsapp_number: string | null;
  instagram_url: string | null;
  website_url: string | null;

  base_price: string;
  bouncy_castle_cost: string;
  extra_hour_price: string;

  bank_name: string | null;
  bank_account_holder: string | null;
  bank_account_number: string | null;
  bank_clabe: string | null;

  facebook_response_message: string | null;
  banking_information_message: string | null;
  location_message: string | null;
  venue_rules_message: string | null;
  payment_reminder_message: string | null;
  thank_you_message: string | null;
  general_notes: string | null;
}

export interface VenueUpdate {
  name: string;
  address: string | null;
  phone: string | null;
  capacity: number;

  opening_time: string | null;
  closing_time: string | null;

  facebook_url: string | null;
  whatsapp_number: string | null;
  instagram_url: string | null;
  website_url: string | null;

  base_price: number;
  bouncy_castle_cost: number;
  extra_hour_price: number;

  bank_name: string | null;
  bank_account_holder: string | null;
  bank_account_number: string | null;
  bank_clabe: string | null;

  facebook_response_message: string | null;
  banking_information_message: string | null;
  location_message: string | null;
  venue_rules_message: string | null;
  payment_reminder_message: string | null;
  thank_you_message: string | null;
  general_notes: string | null;
}